django/test/testcases.py

```diff
@@ -730,6 +731,22 @@ def assertWarnsMessage(self, expected_warning, expected_message, *args, **kwargs
             *args, **kwargs
         )
 
+    @contextmanager
+    def assertNoLogs(self, logger_name, levelname='INFO'):
+        """
+        Assert no logs were emitted by logger_name in wrapped command.
+
+        https://bugs.python.org/issue39385
+        """
+        if isinstance(levelname, int):
+            levelname = logging.getLevelName(levelname)
+        try:
+            with self.assertLogs(logger_name, levelname):
+                yield
+        except AssertionError as e:
+            msg = e.args[0]
+            self.assertEqual(msg, 'no logs of level %s or higher triggered on %s' % (levelname, logger_name))
```

Comment:
We could use unittest's `assertNoLogs()` on Python 3.10+, and add a comment that out implementation should be removed when dropping support for Python 3.9.